config/options.go

```diff
@@ -28,7 +28,7 @@ type options struct {
 // WithSource with config source.
 func WithSource(s ...Source) Option {
 	return func(o *options) {
-		o.sources = s
+		o.sources = append(o.sources, s...)
```

Comment:
Maybe the original `WithSource` is enough？

Reply:
User maybe use WithSource many times with different Source

Reply:
`WithSource` can accept multiple parameters, what are the scenarios that require the use of multiple `WithSource`?

Reply:
User don't know WithSource can only use once with all sources, they maybe use twice and more with different sources. The framework should be comfortable for users to use.